helios-services/src/main/java/com/spotify/helios/master/DeadAgentReaper.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2016 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.helios.master;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Lists;
+
+import com.spotify.helios.agent.InterruptingScheduledService;
+import com.spotify.helios.common.Clock;
+import com.spotify.helios.common.SystemClock;
+import com.spotify.helios.common.descriptors.AgentInfo;
+import com.spotify.helios.common.descriptors.HostStatus;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * De-registers dead agents, where an agent that has been DOWN for more than X time is considered
+ * dead.
+ */
+public class DeadAgentReaper extends InterruptingScheduledService {
+
+  private static final Clock SYSTEM_CLOCK = new SystemClock();
+  private static final long INTERVAL = 30;
+  private static final TimeUnit INTERVAL_TIME_UNIT = TimeUnit.MINUTES;
```

Comment:
I think that might really surprise someone if we were to ever actually reach uptime of 3 years :smirk: 

I don't expect to have to change it often but if there ever is it is nicer IMHO to have the option to change configuration rather than re-releasing Helios (and have the default value be the 30 min one here). 

I think the clutter of adding new CLI args has more to do with how we've implemented it with the Config, Parser, Server etc. I am curious if we would get more mileage out of just reading some sort of yaml/json file at startup and not having to have any new switch/flag be added in 3-4 places.